src/MiddlewarePipeInterface.php

```diff
@@ -10,4 +10,7 @@
 interface MiddlewarePipeInterface extends MiddlewareInterface, RequestHandlerInterface
 {
     public function pipe(MiddlewareInterface $middleware): void;
+
+    /** @return list<MiddlewareInterface> */
+    public function toArray(): array;
```

Comment:
IMO not to be interfaced :thinking:

Reply:
I thought so too… If you need the convenience of inspecting the pipeline, like in tests, then it's not much to ask to first assert the implementation type.

Reply:
@gsteel what about @Xerkus' suggestion in #44? Or even yours?

Reply:
I've added a new interface with no additional methods that extends from `MiddlewarePipeInterface` and `IteratorAggregate` so that users can test for `IterableMiddlewarePipeInterface` and simply call `iterator_to_array` on the pipeline instance. Better?